### PR TITLE
Add cmake files to support Tizen gbs build

### PR DESCRIPTION
**Pull request by haesik**
libtuv-DCO-1.0-Signed-off-by: Haesik, Jun haesik.jun@samsung.com

### CODE REVIEW

File: cmake/config/config_gbs-tizen.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(CMakeForceCompiler)
+
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR arm)
```
Comment (glistening) [inline cmake/config/config_gbs-tizen.cmake]: @haesik We use the form of "_{architecture}-{os}_.cmake". What about `config_arm-tizen-gbs`?

File: cmake/option/option_gbs-tizen.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          ${FLAGS_COMMON}
+          "-mfloat-abi=softfp"
+          "-mfpu=neon-vfpv4"
+          )
+  else()
+    message(FATAL_ERROR "TARGET_BOARD=`${TARGET_BOARD}` is unknown to make")
+  endif()
+else()
+  message(FATAL_ERROR "TARGET_BOARD is undefined")
+endif()
```
Comment (glistening) [inline cmake/option/option_gbs-tizen.cmake]: @haesik What about `option_arm-tizen-gbs.cmake`?

File: cmake/config/config_gbs-tizen.cmake (hunk 1 above)
Comment (haesik) [reply cmake/config/config_gbs-tizen.cmake]: I don't agree with your opinion.. Currently, considering gbs as an architecture is now easy to support gbs build.. if we uas config_arm-tizen-gbs, we should make config_arm-tizen-gbs, config_x86-tizen-gbs, config_x86_64-tizen-gbs, etc,,.. for the same gbs build.

File: cmake/option/option_gbs-tizen.cmake (hunk 2 above)
Comment (haesik) [reply cmake/option/option_gbs-tizen.cmake]: I don't agree with your opinion.. Currently, considering gbs as an architecture is now easy to support gbs build.. if we uas config_arm-tizen-gbs, we should make config_arm-tizen-gbs, config_x86-tizen-gbs, config_x86_64-tizen-gbs, etc,,.. for the same gbs build.

Comment (glistening) [reply cmake/option/option_gbs-tizen.cmake]: @haesik I also know considering `gbs` as an architecture may be easiest way to get it working. But `gbs` is not an architecture. And it is open source that many developers work on same code base. I don't want to break the rule for reducing work on tizen. Alternatively, we may use `option_noarch-tizen-gbs.cmake` if a common file can be used for multiple targets.

Comment (haesik) [reply cmake/option/option_gbs-tizen.cmake]: I think you don't want to consider gbs as architecture.. I agree to use noarch for gbs.
the file name could be option_noarch-tizen.cmake

**Pull request merged**